### PR TITLE
Upgrade Helm to v3.6.3 and use apiextensions-application

### PR DESCRIPTION
**Pull request by rossf7**
Towards giantswarm/giantswarm#19819

## Checklist

- [x] Update changelog in CHANGELOG.md.

### CODE REVIEW

File: service/controller/chart/key/key.go [hunk 1]
```diff
@@ -94,7 +94,7 @@ func TarballURL(customResource v1alpha1.Chart) string {
 	return customResource.Spec.TarballURL
 }
 
-// ToCustomResource converts value to v1alpha1.ChartConfig and returns it or error
+// ToCustomResource converts value to v1alpha1.Chart and returns it or error
```
Comment (rossf7) [inline service/controller/chart/key/key.go:97]: Just fixing typo

File: service/controller/chart/resource/release/delete_test.go [hunk 2]
```diff
@@ -15,7 +15,7 @@ import (
 
 func Test_Resource_Release_newDeleteChange(t *testing.T) {
 	testCases := []struct {
-		obj                  v1alpha1.ChartConfig
+		obj                  v1alpha1.Chart
```
Comment (rossf7) [inline service/controller/chart/resource/release/delete_test.go:18]: ChartConfig CRD is ancient but somehow this didn't get updated 😬

File: go.mod [hunk 3]
```diff
@@ -3,30 +3,35 @@ module github.com/giantswarm/chart-operator/v2
...
-	github.com/giantswarm/apiextensions/v3 v3.32.0
-	github.com/giantswarm/app/v5 v5.1.0
+	github.com/giantswarm/apiextensions-application v0.1.0
+	github.com/giantswarm/app/v6 v6.0.0
 	github.com/giantswarm/appcatalog v0.6.0
 	github.com/giantswarm/backoff v0.2.0
 	github.com/giantswarm/exporterkit v0.2.1
-	github.com/giantswarm/helmclient/v4 v4.8.0
-	github.com/giantswarm/k8sclient/v5 v5.12.0
+	github.com/giantswarm/helmclient/v4 v4.8.1-0.20211202155511-0bfbea24b901
```
Comment (rossf7) [inline go.mod:13]: Needs https://github.com/giantswarm/helmclient/pull/385

File: integration/setup/setup.go [hunk 4]
```diff
@@ -105,7 +103,7 @@ func installResources(ctx context.Context, config Config) error {
 			return microerror.Mask(err)
 		}
 
-		err = config.K8sClients.CRDClient().EnsureCreated(ctx, chartCRD, backoff.NewMaxRetries(7, 1*time.Second))
+		err = config.K8sClients.CtrlClient().Create(ctx, chartCRD)
```
Comment (MarcelMue) [inline integration/setup/setup.go:106]: I assume we can't configure the backoff here?

**Review by MarcelMue — COMMENTED** [review]
Should we wait for the helmclient merge?

File: integration/setup/setup.go (hunk 4 above)
Comment (rossf7) [reply integration/setup/setup.go:106]: Yes the CRD package in k8sclient had retry logic which was configured here. It was removed in k8sclient v6.

**Comment by rossf7** [comment]
> Should we wait for the helmclient merge?

@MarcelMue Yes, I want to have the PRs prepared but I'm not planning on releasing helmclient until the app-operator changes are ready which is the last component.

**Review by rossf7 — COMMENTED** [review]
Ping @giantswarm/team-honeybadger 

This is ready now PTAL 🙏

**Review by MarcelMue — APPROVED** [review]

**Pull request merged**